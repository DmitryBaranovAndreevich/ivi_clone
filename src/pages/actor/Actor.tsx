import { Link, useNavigate, useParams } from 'react-router-dom';
import Spinner from '../../components/UI/spinner/Spinner';
import { personApi } from '../../store/api/personApi';
import styles from './actor.module.scss';
import Line from '../../components/login/Line/Line';
import Movie from '../../components/UI/Movie/Movie';
import { useTranslation } from 'react-i18next';
import LinkBack from '../../components/UI/linkStepBack/LinkBack';

const Actor = () => {
  const { t } = useTranslation();
  const { id } = useParams();
  const { data, isLoading } = personApi.useGetOnePersonQuery({ id });
  const navigate = useNavigate();
  return isLoading ? (
    <div className={styles.spinner}>
      <Spinner size={'big'} />
    </div>
  ) : (
    <div className={styles.wrapper}>
      <LinkBack href={`#`} onClick={() => navigate(-1)} text={t('actor.commBack')} />
      <div className={styles.container}>
        <img src={data?.photo} alt="actor foto" className={styles.foto} />
        <h1 className={styles.rusName}>{data?.name}</h1>
        <h3 className={styles.engName}>{data?.originalName}</h3>
        <p className={styles.title}>
          {t('actor.films')}
          <span className={styles.span}>
            {t('actor.quantity')}
            {data?.films.length}
          </span>
        </p>
        <div>
          <p className={styles.smallTitle}>{t('actor.actor')}</p>
          <Line persent={'0'} />
        </div>
        {data?.films.map((movie) => (
          <Movie data={movie} key={movie.id} />
        ))}
      </div>
    </div>
  );
};

export default Actor;
