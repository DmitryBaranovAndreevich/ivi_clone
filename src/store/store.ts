import reviewReducer from './reducers/ReviewSlice';
import moviesSortReduser from './reducers/MoviesSort';
import { appApi } from './api/appApi';
import { combineReducers, configureStore } from '@reduxjs/toolkit';
import userLoginReduser from './reducers/UserLoginSlice';
import appReducer from './reducers/App';
import moviesFilterReduser from './reducers/MoviesFilter';
import { filmApi } from './api/filmApi';
import { personApi } from './api/personApi';
import { reviewApi } from './api/reviewApi';

const rootReducer = combineReducers({
  appReducer,
  userLoginReduser,
  moviesSortReduser,
  reviewReducer,
  moviesFilterReduser,
  [appApi.reducerPath]: appApi.reducer,
  [filmApi.reducerPath]: filmApi.reducer,
  [personApi.reducerPath]: personApi.reducer,
  [reviewApi.reducerPath]: reviewApi.reducer,
});

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat([appApi.middleware, filmApi.middleware, personApi.middleware]),
  });
};

export type TRootState = ReturnType<typeof rootReducer>;
export type TAppStore = ReturnType<typeof setupStore>;
export type TAppDispatch = TAppStore['dispatch'];
