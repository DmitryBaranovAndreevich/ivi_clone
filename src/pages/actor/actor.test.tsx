import { Provider } from 'react-redux';
import { store } from '../../store/mainStore';
import renderer from 'react-test-renderer';
import * as router from 'react-router';
import { MemoryRouter } from 'react-router-dom';
import Actor from './Actor';

describe('Тестим страницу входа', () => {
  const navigate = jest.fn();

  jest.mock('react-i18next', () => ({
    useTranslation: jest.fn(),
  }));

  beforeEach(() => {
    jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
  });
  it('Страница loginEmail рендерится без ошибок', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Actor />
        </MemoryRouter>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
