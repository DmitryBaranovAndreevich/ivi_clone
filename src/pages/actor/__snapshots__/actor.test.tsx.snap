// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестим страницу входа Страница loginEmail рендерится без ошибок 1`] = `
<div>
  <div
    className="{spinnerStyle.spinner} undefined"
  >
    <i
      className="undefined undefined"
    />
  </div>
</div>
`;
